lib/wallaroo/core/topology/steps.pony

```diff
@@ -156,9 +197,22 @@ actor Step is (Producer & Consumer)
     _omni_router = omni_router
 
     _initialized = true
-    initializer.report_initialized(this)
 
   be application_initialized(initializer: LocalTopologyInitializer) =>
+    // _initializer = initializer
+    // if _routes.size() > 0 then
+    //   for r in _routes.values() do
+    //     r.application_initialized("Step")
+    //   end
+    // else
+    //   _report_ready_to_work()
+    // end
+    _initializer_initialized(initializer)
+
+  be initializer_initialized(initializer: _ReportReadyToWork) =>
+    _initializer_initialized(initializer)
+
+  fun ref _initializer_initialized(initializer: _ReportReadyToWork) =>
```

Comment:
rename `_prepare_report_ready_to_work`.

Reply:
Use sum type instead of `_ReportReadyToWork`

Reply:
done